tezos/protocol-ipc-client/src/lib.rs

```diff
@@ -90,54 +94,68 @@ impl ProtocolRunnerConfiguration {
 
 // TODO: differentiate between writable and readonly runners?
 
-/// Representation of a running protocol runner instance.
-pub struct ProtocolRunnerInstance {
-    configuration: ProtocolRunnerConfiguration,
-    socket_path: PathBuf,
-    endpoint_name: String,
-    tokio_runtime: tokio::runtime::Handle,
-    child_process_handle: Option<Child>,
-    context_stats_db_path: Option<PathBuf>,
-    patch_context: Option<PatchContext>,
-    shutdown_issued: bool,
-    log: Logger,
+struct IpcIO {
+    rx: IpcReceiver<NodeMessage>,
+    tx: IpcSender<ProtocolMessage>,
 }
 
-impl Drop for ProtocolRunnerInstance {
-    fn drop(&mut self) {
-        self.shutdown()
+impl IpcIO {
+    pub async fn send(&mut self, value: &ProtocolMessage) -> Result<(), async_ipc::IpcError> {
+        self.tx.send(value).await?;
+        Ok(())
+    }
+
+    pub async fn try_receive(
+        &mut self,
+        read_timeout: Option<Duration>,
+    ) -> Result<NodeMessage, async_ipc::IpcError> {
+        let result = if let Some(read_timeout) = read_timeout {
+            self.rx.try_receive(read_timeout).await?
+        } else {
+            self.rx.receive().await?
+        };
+        Ok(result)
     }
 }
 
-impl ProtocolRunnerInstance {
-    pub const PROCESS_TERMINATE_WAIT_TIMEOUT: Duration = Duration::from_secs(10);
+/// Manages the execution of the protocol runners and access to their functionality.
```

Comment:
Remember that we want to be able to support multiple protocol runners, this combination of `ProtocolRunnerInstance` and `ProtocolRunnerApi` doesn't play well with that.

Reply:
Well, it all depends on how it ends being represented in the state machine. We probably will have many instances of this struct inside the state machine, right? with the connections explicitly handled there (instead of being asked-for by the caller like it works in the actors system right now, which is not a great way to do it).